src/entity-manager/MongoEntityManager.ts

```diff
@@ -744,4 +770,8 @@ export class MongoEntityManager extends EntityManager {
         };
     }
 
+    protected filterSoftDeleted<Entity>(cursor: Cursor<Entity>, deleteDateColumn: ColumnMetadata) {
+        cursor.filter({$where: `this.${deleteDateColumn.propertyName}==null`});
```

Comment:
Does `filter` mutate the original `cursor` ?

Reply:
It does.
```js
Cursor.prototype.filter = function(filter) {
  if(this.s.state == Cursor.CLOSED || this.s.state == Cursor.OPEN || this.isDead()) throw MongoError.create({message: "Cursor is closed", driver:true});
  this.s.cmd.query = filter;
  return this;
}
```
But that works only for a single entity with no relations.